internal/querycoordv2/observers/handoff_observer.go

```diff
@@ -230,8 +230,10 @@ func (ob *HandoffObserver) tryHandoff(ctx context.Context, segment *querypb.Segm
 
 	log.Info("try handoff segment...")
 	status, ok := ob.collectionStatus[segment.GetCollectionID()]
+	_, added := ob.handoffEvents[segment.GetSegmentID()]
 	if Params.QueryCoordCfg.AutoHandoff &&
```

Comment:
doesn't make sense to me,
see the code here:
		case <-ticker.C:
			for _, event := range ob.handoffEvents {
				switch event.Status {
				case HandoffEventStatusReceived:
					ob.tryHandoff(ctx, event.Segment)
				case HandoffEventStatusTriggered:
					ob.tryRelease(ctx, event)
				}
			}

			ob.tryClean(ctx)
		}